BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c

```diff
@@ -542,7 +545,7 @@ FindPartitions (
 
   CopyMem (&PartBlockDev->BlockInfo, &DevBlockInfo, sizeof (DEVICE_BLOCK_INFO));
 
-  if (DevBlockInfo.BlockSize == 1) {
+  if (mCurrentMediaType == OsBootDeviceMemory || mCurrentMediaType == OsBootDeviceMemory) {

```

Comment:
I guess you meant OsBootDeviceSpi.
```
if (mCurrentMediaType == OsBootDeviceMemory || mCurrentMediaType == OsBootDeviceMemory)
->
if ((CurrentMediaType == OsBootDeviceSpi) || (CurrentMediaType == OsBootDeviceMemory))
```

Quick question. Did the original code 'if (DevBlockInfo.BlockSize == 1)' cause a problem? Do we really need this change?

Reply:
Using BlockSize==1 to check SPI/Mem device will work as long as you set block size to 1. But for SPI, we originally have bigger block size.  I think it is always good to use the actual media type for check.

Reply:
OsBootDeviceSpi is mainly used for PDR region. For Memory mapped SPI BIOS region, OsBootDeviceMemory should be used for better performance. What issue for this patch is to solve?

Reply:
OsBootDeviceSpi is supported for BIOS region also now. This patch if for optimization and improving the code checks.

Reply:
Understood. So, this is code improvement to check actual media type. Please address review comments.